### DISCUSSION TITLE
What stats does the speed of the prefetching algorithm affect?

### DISCUSSION THREAD

**Discussion by afschy** [ChampSim/ChampSim#481] [Q&A] [answered]
If I wrote a prefetching algorithm for L2C that does a lot of computation upon a demand access or eviction, will it have an effect on average miss latency for L2C, cycle count, or any other stat? Or is the simulation agnostic towards the performance of the prefetcher?

**Comment by ngober** [comment, marked as answer]
There is no method by which ChampSim detects how long your algorithm runs on the host machine. You can perform pipelined operations in the `prefetcher_cycle_operate()` hook.
